src/axom/quest/IntersectionShaper.hpp

```diff
@@ -1466,6 +1441,454 @@ class IntersectionShaper : public Shaper
+        {
+          continue;
+        }
+
+        // both Z are (nearly) the same
+        if(isNearlyEqual(cur_point[Z], prev_point[Z], EPS))
+        {
+          continue;
+        }
+
+        // both points are (nearly) the same
+        if(squared_distance(cur_point, prev_point) < EPS_SQ)
+        {
+          continue;
+        }
+      }
+
+      // Add the current point to the new mesh.
+      newm->appendNode(cur_point[Z], cur_point[R]);
+      prev_point = cur_point;
+      polyline_size += 1;
+    }
+
+    // Get the ZR coordinates.
+    double* z = newm->getCoordinateArray(mint::X_COORDINATE);
+    double* r = newm->getCoordinateArray(mint::Y_COORDINATE);
+
+    // Check if we need to flip the points order.
+    // discretize() is only valid if x increases as index increases
+    bool flip = false;
+    if(polyline_size > 1)
+    {
+      if(z[0] > z[1])
+      {
+        flip = true;
+        SLIC_INFO("Order of contour points has been reversed!"
+                  << " Discretization algorithm expects Z values of contour "
+                     "points to be increasing");
+      }
+    }
+
+    SLIC_INFO(axom::fmt::format(
+      "{:-^80}",
+      axom::fmt::format(" Discretizing contour with {} points ", polyline_size)));
+
+    // Flip point order
+    if(flip)
+    {
+      int i = polyline_size - 1;
+      int j = 0;
+      while(i > j)
+      {
+        axom::utilities::swap(z[i], z[j]);
+        axom::utilities::swap(r[i], r[j]);
+        i -= 1;
+        j += 1;
+      }
+    }
+
+    // Now make polyline_size - 1 segments.
+    int numNewSegments = polyline_size - 1;
+    newm->reserveCells(numNewSegments);
+    for(int i = 0; i < numNewSegments; ++i)
+    {
+      IndexType seg[2] = {i, i + 1};
+      newm->appendCell(seg, mint::SEGMENT);
+    }
+
+    return newm;
+  }
+
+  /*!
+   * \brief Compute the area of the polygon given by \a pts.
+   * \param pts The list of points that make up the polygon.
+   * \return The polygon area.
+   */
+  double area(const std::vector<Point2D>& pts) const
+  {
+    auto npts = static_cast<int>(pts.size());
+    // Compute the areas
+    double A = 0.;
+    for(int i = 0; i < npts; i++)
+    {
+      int nexti = (i + 1) % npts;
+      A += pts[i][0] * pts[nexti][1] - pts[i][1] * pts[nexti][0];
+    }
+    // Take absolute value just in case the polygon had reverse orientation.
+    return fabs(A * 0.5);
+  }
+
+  /*!
+   * \brief Compute the circle area of a circle of radius at \a level.
+   * \param radius The radius of the circle.
+   * \return The circle area
+   */
+  double calcCircleArea(double radius, int level) const
+  {
+    int npts = 3 * pow(2, level);
+    std::vector<Point2D> pts;
+    pts.reserve(npts);
+    for(int i = 0; i < npts; i++)
+    {
+      double angle =
+        2. * M_PI * static_cast<double>(i) / static_cast<double>(npts);
+      pts.push_back(Point2D {radius * cos(angle), radius * sin(angle)});
+    }
+    return area(pts);
```

Comment:
It might be worth mentioning that this is a discretized circle defined by `npts` linear segments on its boundary.

Reply:
I added a comment.